MapboxGeocoder/MBGeocoder.swift

```diff
@@ -223,7 +223,15 @@ open class Geocoder: NSObject {
             let decoder = JSONDecoder()
             
             do {
-                let result = try decoder.decode([GeocodeResult].self, from: data)
+                let result: [GeocodeResult]
+                do {
+                    // Decode multiple batch geocoding queries
+                    result = try decoder.decode([GeocodeResult].self, from: data)
+                } catch {
+                    // Decode single batch geocding queries
+                    result = [try decoder.decode(GeocodeResult.self, from: data)]
```

Comment:
As tail work, maybe we should consider refactoring Geocoder’s public methods so that we distinguish between single-serving and batched requests at different places in the call stack.